### PR TITLE
Node: bug fixes for the FB reverse geo code and the required fields completion

### PR DESCRIPTION
**Pull request by mbadreldin**
Adding logic to not trust street address on reverse geo coding. Fixing the unknown FB locations crash. Making sure formatted addresses are updated on required fields completion.

### CODE REVIEW

File: Node/core/src/dialogs/require-fields-dialog.ts [hunk 1]
```diff
@@ -45,6 +45,7 @@ function createDialog() {
 
                 session.dialogData.lastInput = session.message.text;
                 session.dialogData.place.address[fields[index].name] = session.message.text;
+                session.dialogData.place.address.formattedAddress = common.getFormattedAddressFromLocation( session.dialogData.place, ", ");
```
Comment (kgrashad) [inline Node/core/src/dialogs/require-fields-dialog.ts]: We read this separator from localized strings, no?

File: Node/core/src/dialogs/retrieve-facebook-location-dialog.ts [hunk 2]
```diff
@@ -19,9 +19,20 @@ function createDialog(apiKey: string) {
...
                     .then(locations => {
-                        var place: RawLocation;
+                        let place: RawLocation;
                         if (locations.length) {
-                            place = locations[0];
+                            // We don't trust reverse geo-coder on the street address level.
+                            // So, copy all fields except it.
+                            let address: Address = {
+                                addressLine : undefined,
+                                formattedAddress: locations[0].address.formattedAddress,
```
Comment (kgrashad) [inline Node/core/src/dialogs/retrieve-facebook-location-dialog.ts]: Why are we using the formattedAddress coming from Bing? We should not trust this one also, right?

File: Node/core/src/dialogs/retrieve-facebook-location-dialog.ts [hunk 3]
```diff
@@ -19,9 +19,20 @@ function createDialog(apiKey: string) {
             if (session.dialogData.args.reverseGeocode && results.response && results.response.place) {
                 locationService.getLocationByPoint(apiKey, results.response.place.point.coordinates[0], results.response.place.point.coordinates[1])
                     .then(locations => {
-                        var place: RawLocation;
+                        let place: RawLocation;
                         if (locations.length) {
```
Comment (kgrashad) [inline Node/core/src/dialogs/retrieve-facebook-location-dialog.ts]: add && locations[0].address

Comment (mbadreldin) [inline Node/core/src/dialogs/retrieve-facebook-location-dialog.ts]: good catch

**Pull request merged**